inmem_endpoint.go

```diff
@@ -129,3 +134,45 @@ func formatSamples(source map[string]SampledValue) []SampledValue {
 
 	return output
 }
+
+// Stream writes metrics to resp each time an interval ends. Runs until the
+// request context is cancelled.
+func (i *InmemSink) Stream(resp http.ResponseWriter, req *http.Request) {
+	interval := i.getInterval()
+	ctx := req.Context()
+
+	resp.WriteHeader(http.StatusOK)
+	flusher, ok := resp.(http.Flusher)
+	if ok {
+		// call Write with 0 bytes before a flush, so that GzipResponseWriter
+		// can write response headers.
+		resp.Write([]byte(""))
+		flusher.Flush()
+	} else {
+		flusher = noopFlusher{}
+	}
+
+	encoder := json.NewEncoder(resp)
+
+	for {
+		select {
+		case <-interval.done:
+			summary := newMetricSummaryFromInterval(interval)
+
+			if err := encoder.Encode(summary); err != nil {
+				// TODO: pass in a logger to log this error
```

Comment:
I almost forgot about this TODO. I guess this would be good. 

Maybe the args to this function should be a `context.Context` and a logger, instead of the `http.Request`.